lib/basehandler.py

```diff
@@ -2,9 +2,47 @@
 from tornado import gen
 
 from .database.auth import deny
+from lib.config import CONFIG
 
 import json
 import time
+import base64
+import bcrypt
+
+
+def secured(handler_class):
+
+    def wrap_execute(handler_execute):
+        def require_basic_auth(handler, kwargs):
+            auth_header = handler.request.headers.get('Authorization')
+            if auth_header is None or not auth_header.startswith('Basic '):
+                handler.set_status(401)
+                handler.set_header('WWW-Authenticate',
+                                   'Basic realm=Restricted')
+                handler._transforms = []
+                handler.finish()
+                return False
+            auth_decoded = base64.b64decode(auth_header[6:]).decode('utf-8')
+            user, password = auth_decoded.split(':', 2)
+            hashed_password = CONFIG.get('admin_user_pass').encode('utf-8')
+            encoded_password = password.encode('utf-8')
+            result = (user == CONFIG.get('admin_user_id') and
```

Comment:
I think this would be more clear if you did:

```
if user == CONFIG.get('admin_user_id') and bcrypt.hashpw(encoded_password, hashed_password) == hashed_password:
    return True
handler.set_status(401)
handler._transforms = []
handler.finish()
return False
```